test/utils/session-core.test.js

```diff
@@ -0,0 +1,249 @@
+import {
+  getSolidDataset,
+  createContainerAt,
+  mockSolidDatasetFrom,
+  mockThingFrom,
+  createSolidDataset,
+  setThing,
+  saveSolidDatasetInContainer
+} from '@inrupt/solid-client';
+import { expect, vi, it, describe, afterEach, beforeEach } from 'vitest';
+import { getContainerUrl } from '../../src/utils/network/session-helper';
+import { INTERACTION_TYPES } from '../../src/constants';
+
+vi.mock('@inrupt/solid-client', async () => {
+  const actual = await vi.importActual('@inrupt/solid-client');
+  return {
+    ...actual,
+    getSolidDataset: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url))),
+    createContainerAt: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url))),
+    saveSolidDatasetInContainer: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url)))
+  };
+});
+
+const getSolidDatasetRejected = vi.fn(() => Promise.reject(Error('dataset not found')));
+
+const mockPodUrl = 'https://pod.example.com/';
+const mockSolidIdentityProvider = 'https://example.com/';
+let session = {};
+
+describe('setDocAclPermission', () => {
+  beforeEach(() => {
+    session = {
+      fetch: vi.fn(),
+      info: {
+        webId: `${mockPodUrl}profile/card#me`
+      }
+    };
+  });
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it("prints out correct documentUrl for Driver's License and webId for other user", () => {
+    const fileType = "Driver's License";
+    const otherPodUsername = 'pod2';
+    const documentUrl = getContainerUrl(session, fileType, INTERACTION_TYPES.SELF);
+    const webId = `https://${otherPodUsername}.${
```

Comment:
You generally shouldn't need complex string parsing like this in unit tests. If there's a string you want to declare, just declare it.

`const webId = 'https://pod2.example.com/profile/card#me'`

Or, since our system doesn't actually care about webId format:
`const webId = \`${mockSolidIdentityProvider}pod2\``

Reply:
Wanted to see if the parsing is correct for that test case (it's in the original function). If webId comes out wrong, it'll error out.

Reply:
If the code is in the original function, then you should generate the ID by calling that function.